src/pages/profile/[teacherId].tsx

```diff
@@ -25,19 +26,27 @@ import { getExperienceByTeacherId, getRatings, getTeacherByTeacherId } from '@/a
 export async function getServerSideProps(context: any) {
     const teacherId = context.params.teacherId;
 
-    const [
-        teacher,
-        experience,
-        ratings
-    ] = await Promise.all([
-        getTeacherByTeacherId(teacherId),
-        getExperienceByTeacherId(teacherId),
-        getRatings(teacherId),
-    ]);
-
-    return {
-        props: { teacher, experience, ratings },
-    };
+    try {
+        const [
+            teacher,
+            experience,
+            ratings
+        ] = await Promise.all([
+            getTeacherByTeacherId(teacherId),
+            getExperienceByTeacherId(teacherId),
+            getRatings(teacherId),
+        ]);
+
+        return {
+            props: { teacher, experience, ratings },
+        };
+    } catch (error) {
+        if (error instanceof AxiosError && error.status === 404) {
```

Comment:
What should be happening when status is other than 404?

Reply:
adjusted